### ISSUE TITLE
Voice Broadcasts: Clarity on multi broadcast across multi VoiceConnection Objects

### ISSUE THREAD

**Issue by KM0KA** [discordjs/discord.js#3960]
**Please describe the problem you are having in as much detail as possible:**

I have been able to set up and play Broadcasts. It was a real struggle getting the environment set up, but now it is stable and I can have an infinite number select a channel to listen to. 

If you can set up multiple Broadcasts, each playing a different audio source, and then have that broadcast connect to multiple connections, with each connection tying back to a listener, how does one achieve that, to have one single Bot App create multiple Broadcasts each playing their own stream and have that Bot/App "join" to get a connection, without dropping its current connection.

Let me explain:

I understand to get the VoiceConnection object, one does a:

member.voice.channel.join()
.then (connection => {

This action immediately will disconnect the Bot from its current channel and establish a connection with the new channel. Not what one would want in a multi broadcast configuration.

The docs state the following:

// Create an instance of a VoiceBroadcast:

const broadcast1 = client.voice.createBroadcast();
const broadcast2 = client.voice.createBroadcast();
const broadcast3 = client.voice.createBroadcast();

// Play audio on the broadcast:

const dispatcher1 = broadcast1.play('audio1.mp3');
const dispatcher2 = broadcast2.play('audio2.mp3');
const dispatcher3 = broadcast3.play('audio3.mp3');

// Play this broadcast across multiple connections (subscribe to the broadcast):

connection1.play(broadcast1);
connection2.play(broadcast2);
connection3.play(broadcast3);

I am able to use the client.on('voiceStateUpdate', (oldState, newState) => { to know when a user clicks on a Voice Channel. From here I can get everything . . . but . . . the VoiceConnection . . . 

So that is my question . . . how  does one get multiple connections like above to play:

connection1.play(broadcast1);
connection2.play(broadcast2);
connection3.play(broadcast3);

Without joining the channel, there by causing a disconnect from the old and a connect to the new channel . . . 

Thanks in advance . . . 

**Include a reproducible code sample here, if possible:**

```js
// Place your code here
```

**Further details:**

- discord.js version:
- Node.js version:
- Operating system:
- Priority this issue should have – please be realistic and elaborate if possible:

- [ ] I have also tested the issue on latest master, commit hash:

**Comment by LxveFades** [comment]
I'm not quite sure on what you're trying to ask here, but I'm pretty sure that you're asking on how a Client instance can broadcast audio across multiple connections. And, that's not possible.
Bots are allowed a single voice connection per guild, and users have a single voice connection **globally** (if I recall correctly). 
My point being, you can dispatch audio to a single voice connection per guild. A client cannot be in more than a single voice channel (again, per guild).

Again, I may have misinterpreted what you're trying to say. But I don't think that this is a bug, however I could be wrong, which depends on whether or not I understood what you were trying to say.

**TL;DR**: Bots have a single voice connection per guild; isn't a library bug, that is if I understood the issue's context correctly.

**Comment by KM0KA** [comment]
Thanks @LxveFades.

Then what is the purpose of being able to have multiple connections as shown in the Broadcast examples?

I can have one Voice Channel and an infinite amount of people can join simply by clicking on it, and I can be broadcasting an audio file, and everyone can hear. I have done this with multiple devices. A one to many configuration . . . which is what the old way of handling voice was about . . . 

I have done this doing it the old way and with the new Broadcasting suite of tools.

But as stated above, you can have multiple broadcasts, playing different audio streams, connected to different multiple connections.

My understanding was/is the same as yours, that a Bot/Client can had one voice connection . . . but that causes my "non compute" engine to go off because it conflicts with what the docs states is possible using Broadcasts . . . 

That is, according to the V12 Voice documentation, which I what I have installed, V12 on a NodeJs 13.10 environment . . .

**Comment by KM0KA** [comment]
and you are right, not a bug . . . as it is more confusion with the gap that exists between creating the broadcast and handling multiple connections . . .

**Comment by LxveFades** [comment]
[This is off-topic]
I'm sorry, but I cannot help here. 
My understanding of how the `Voice` part of Discord.js works is still unclear to me, and that territory is quite unknown to me to begin with.
I can't say for certain if you should move to the Discord Server, or perhaps wait for someone who's more experienced (or someone who knows more about how Voice works in Discord.js) to come and for them to give their answer/suggestion.

In both cases, good luck with the issue you're currently facing.

**Comment by amishshah** [comment]
A broadcast is simply a driver for multiple connections - it allows you to play the same audio resource across multiple connections.

I'm not exactly sure what the issue is here? Could you provide some more clarification? It sounds like there's a misconception about how connecting to voice channels work.

An example use-case might be a radio bot that plays the same stream across all servers.

What has been stated above is correct; a bot is allowed to be connected to one voice channel per guild, and users are allowed to connect to one voice channel globally.

**Comment by KM0KA** [comment]
@amishshah 

This single statement "a bot is allowed to be connected to one voice channel per guild," turned on the light for me.

I get it now.

A Bot can handle multiple connections across multiple guilds . . . I saw the one to many connection instantly when i read your statement above. If there are a million guilds, each one can have a channel and that same broadcast can be simulcast . . . 

Closed.
